### PR TITLE
httpserver extension

### PR DESCRIPTION
**Pull request by lmangani**
Drafting a PR for the experimental DuckDB httpserver extension

### CODE REVIEW

**Comment by carlopi** [comment]
I think I fixed the problem you bumped on in CI, can you try merging with main? Thanks

**Comment by lmangani** [comment]
@carlopi not sure what's new but now docker commands are failing....

**Comment by carlopi** [comment]
That should also be fixed now, sorry for the inconvenience, can you give it another try? Thanks a lot

**Comment by lmangani** [comment]
Trying with review mode 👍

File: extensions/httpserver/description.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+extension:
+  name: httpserver
+  description: Simple DuckDB HTTP API extension
+  version: 1.0.0
```
Comment (carlopi) [inline extensions/httpserver/description.yml:4]:
```suggestion
  version: 0.1.0
```
Maybe, given it's experimental?

File: extensions/httpserver/description.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  language: SQL & C++
+  build: cmake
+  license: MIT
```
Comment (carlopi) [inline extensions/httpserver/description.yml:7]:
```suggestion
  license: MIT
  excluded_platforms: "wasm_mvp;wasm_eh;wasm_threads"
```
Unsure what you are using underneath, but I assume this is not there on Wasm. I think it's preferable to just not have it there than having it broken.

Comment (carlopi) [reply extensions/httpserver/description.yml:7]: This is a limit on wasm not running tests (yet).

**Review by carlopi — CHANGES_REQUESTED** [review]
Couple of comments, neither are strong requirements, but please take a look

**Comment by carlopi** [comment]
@lmangani: I took the liberty of merging both minor changes, feel free to reopen a PR reverting them.

File: extensions/httpserver/description.yml (hunk 2 above)
Comment (lmangani) [reply extensions/httpserver/description.yml:7]: Thanks Carlo you're 100% right I doubt the WASM version would be able to open sockets anyway. Perhaps we can think of a websocket based approach if that's an option later on in time.

**Comment by lmangani** [comment]
Thanks @carlopi the changes are spot on and much appreciated.
We'll update the extension ref again today to add the embedded UI and some other features being completed

**Comment by carlopi** [comment]
Should I block the publishing? That's easy on my side.

**Comment by lmangani** [comment]
If its fine to have a 0.0.1 published, it would help us troubleshoot a few real-life scenarios faster. 
I doubt anyone will use it other than for playing around and we'll do fast patches if we find hot issue 👍

**Pull request merged**